tests/gis_tests/geoapp/tests.py

```diff
@@ -510,6 +514,22 @@ def test_subquery_annotation(self):
         )
         self.assertEqual(qs.get(), multifields)
 
+    @skipUnlessDBFeature("supports_distances_lookups", "supports_distance_geodetic")
+    def test_paginating_queryset_with_distance_lookup(self):
+        """
+        Testing that paginating querysets with distance lookups in conditions work.
+        """
+        point = Point(0, 0)
+        cities = City.objects.annotate(
+            relative_distance=Case(
+                When(point__distance_lte=(point, Distance(mi=100)), then=Value(100)),
+                default=Value(1000),
+                output_field=IntegerField(),
+            ),
+        ).order_by('name')
+        page = Paginator(cities, 10).page(1)
+        self.assertIsInstance(page, Page)
```

Comment:
This still crashes when grouping by annotation with `Distance()`, I moved this test to the `gis_tests/distapp`.